core/src/main/clojure/xtdb/operator/scan.clj

```diff
@@ -358,13 +364,16 @@
 
                     :all-time nil)))]
 
-        (apply-constraint for-app-time "application_time_start" "application_time_end")
-        (apply-constraint for-sys-time "system_time_start" "system_time_end"))
+        (apply-constraint for-valid-time "xt$valid_from" "xt$valid_to")
+        (apply-constraint for-system-time "xt$system_from" "xt$system_to"))
 
-      (let [col-types (-> temporal/temporal-col-types (update-keys symbol))
+      (let [ ;; HACK, either in normal form or in datalog form, breaks if mixing conventions
+            col-types (if (and (not-empty selects) (str/includes? (ffirst selects) "/"))
+                        (update-keys temporal/temporal-col-types (comp symbol util/normal-form-str->datalog-form-str))
+                        (update-keys temporal/temporal-col-types symbol))
```

Comment:
Didn't know how to best handle this here. There was one test that broke if I just normalized the columns coming in.
https://github.com/xtdb/core2/blob/7697ef78d8a15802649715c9928435d090f7cea1/src/test/clojure/xtdb/operator/scan_test.clj#L191